spec/models/manageiq/providers/vmware/infra_manager/inventory/collector_spec.rb

```diff
@@ -158,6 +179,51 @@ def vm_delete_object_updates
           ),
         ]
       end
+
+      def vm_new_folder_object_updates
+        [
+          RbVmomi::VIM.ObjectUpdate(
+            :dynamicProperty => [],
+            :kind            => "enter",
+            :obj             => RbVmomi::VIM.Folder(vim, "group-v2000"),
+            :changeSet       => [
+              RbVmomi::VIM.PropertyChange(
+                :dynamicProperty => [],
+                :name            => "name",
+                :op              => "assign",
+                :val             => "test-folder-1",
+              ),
```

Comment:
@agrare - not sure i follow this - are we assigning the value "test-folder-1" to group-v2000 ? if so, why?

Reply:
So a property change contains the name of the property that is being changed, what to do with it, and what the value is.  The property name we are changing is [here](https://github.com/ManageIQ/manageiq-providers-vmware/pull/268/files#diff-351c55bbe0e5544a691437a5a1cb4341R192)

In this case we are assigning the property `name = "test-folder-1"`.

Reply:
This whole [ObjectUpdate](https://github.com/ManageIQ/manageiq-providers-vmware/pull/268/files#diff-351c55bbe0e5544a691437a5a1cb4341R185) is what vcenter would emit when you created a new folder.

Reply:
But why are you changing the name of the folder?

Reply:
It is actually setting the name of the folder for the first time, this is creating a new folder and moving an existing vm into it in one update.

Reply:
Thanks, now it makes sense 👍